src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T [] array = (T[]) new  Object [10];
+    private int pointer;
+
     @Override
     public void add(T value) {
-
+        if (pointer == array.length) {
+            grow(array);
+        }
+        array[pointer++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > pointer) {
```

Comment:
The check for index bounds is repeated in multiple methods. Consider creating a private method to handle index validation and reuse it.